packages/synapse-react-client/src/components/SynapseTable/SynapseTable.tsx

```diff
@@ -160,6 +170,8 @@ export function SynapseTable(props: SynapseTableProps) {
     },
   })
 
+  const { dataHasBeenPrefetched } = usePrefetchTableData()
```

Comment:
Delay rendering the table cells until we have prefetched all the data to prevent the waterfall effect